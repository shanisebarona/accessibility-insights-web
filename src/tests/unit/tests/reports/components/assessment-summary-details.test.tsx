// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';
import {
    AssessmentSummaryDetails,
    AssessmentSummaryDetailsProps,
} from 'reports/components/assessment-summary-details';
import { AssessmentReportBuilderTestHelper } from '../../DetailsView/assessment-report-builder-test-helper';

describe('AssessmentSummaryDetails', () => {
    describe('render', () => {
        test('Correct composition', () => {
            const props: AssessmentSummaryDetailsProps = {
                testSummaries:
                    AssessmentReportBuilderTestHelper.getAssessmentsSummaryReportModel()
                        .reportSummaryDetailsData,
            };

            const wrapper = shallow(<AssessmentSummaryDetails {...props} />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        test('Renders an OutcomeIconSet for 7 items', () => {
            const props: AssessmentSummaryDetailsProps = {
                testSummaries: [
                    {
                        displayName: 'test',
                        pass: 0,
                        fail: 0,
                        incomplete: 7,
                    },
                ],
            };

            const wrapper = shallow(<AssessmentSummaryDetails {...props} />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        test('Renders an OutcomeChipSet for 8 items', () => {
            const props: AssessmentSummaryDetailsProps = {
                testSummaries: [
                    {
                        displayName: 'test',
                        pass: 0,
                        fail: 0,
                        incomplete: 8,
                    },
                ],
            };

            const wrapper = shallow(<AssessmentSummaryDetails {...props} />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });
});
