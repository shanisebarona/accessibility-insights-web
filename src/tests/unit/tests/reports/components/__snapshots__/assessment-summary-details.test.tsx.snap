// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentSummaryDetails render Correct composition 1`] = `
<div
  className="assessmentSummaryDetails"
  role="table"
>
  <div
    className="assessmentSummaryDetailsBody"
    role="rowgroup"
  >
    <div
      className="assessmentSummaryDetailsRow"
      role="row"
    >
      <div
        className="testSummary"
      >
        <div
          className="testSummaryDisplayName"
          role="cell"
        >
          Assessment1
        </div>
        <div
          className="testSummaryStatus"
          data-automation-id="test-summary-status/Assessment1"
          role="cell"
        >
          <OutcomeIconSet
            displayName="Assessment1"
            fail={2}
            incomplete={1}
            pass={1}
          />
        </div>
      </div>
    </div>
    <div
      className="assessmentSummaryDetailsRow"
      role="row"
    >
      <div
        className="testSummary"
      >
        <div
          className="testSummaryDisplayName"
          role="cell"
        >
          Assessment2
        </div>
        <div
          className="testSummaryStatus"
          data-automation-id="test-summary-status/Assessment2"
          role="cell"
        >
          <OutcomeIconSet
            displayName="Assessment2"
            fail={1}
            incomplete={1}
            pass={0}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AssessmentSummaryDetails render Renders an OutcomeChipSet for 8 items 1`] = `
<div
  className="assessmentSummaryDetails"
  role="table"
>
  <div
    className="assessmentSummaryDetailsBody"
    role="rowgroup"
  >
    <div
      className="assessmentSummaryDetailsRow"
      role="row"
    >
      <div
        className="testSummary"
      >
        <div
          className="testSummaryDisplayName"
          role="cell"
        >
          test
        </div>
        <div
          className="testSummaryStatus"
          data-automation-id="test-summary-status/test"
          role="cell"
        >
          <OutcomeChipSet
            displayName="test"
            fail={0}
            incomplete={8}
            pass={0}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AssessmentSummaryDetails render Renders an OutcomeIconSet for 7 items 1`] = `
<div
  className="assessmentSummaryDetails"
  role="table"
>
  <div
    className="assessmentSummaryDetailsBody"
    role="rowgroup"
  >
    <div
      className="assessmentSummaryDetailsRow"
      role="row"
    >
      <div
        className="testSummary"
      >
        <div
          className="testSummaryDisplayName"
          role="cell"
        >
          test
        </div>
        <div
          className="testSummaryStatus"
          data-automation-id="test-summary-status/test"
          role="cell"
        >
          <OutcomeIconSet
            displayName="test"
            fail={0}
            incomplete={7}
            pass={0}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
